src/main/java/org/reactome/addlinks/referencecreators/EnsemblBiomartMicroarrayPopulator.java

```diff
@@ -0,0 +1,192 @@
+     * @param rgpInstances Collection of ReferenceGeneProduct instances
+     * @return Map of RGP identifiers to associated RGP instances
+     * @throws Exception can be caused when retrieving data from GKInstance
+     */
+    private Map<String, ArrayList<GKInstance>> mapRGPIdentifiersToRGPInstances(Collection<GKInstance> rgpInstances) throws Exception {
+        Map<String, ArrayList<GKInstance>> rgpIdentifiersToRGPs = new HashMap<>();
+        for (GKInstance rgpInst : rgpInstances) {
+            String identifier = rgpInst.getAttributeValue(ReactomeJavaConstants.identifier).toString();
+            if (rgpIdentifiersToRGPs.get(identifier) != null) {
+                rgpIdentifiersToRGPs.get(identifier).add(rgpInst);
+            } else {
+                ArrayList<GKInstance> singleRGPIdentifierArray = new ArrayList<>(Arrays.asList(rgpInst));
+                rgpIdentifiersToRGPs.put(identifier, singleRGPIdentifierArray);
+            }
+        }
+        return rgpIdentifiersToRGPs;
+    }
+
+    // Sorts all microarray values added to otherIdentifier slot and updates instance with new data.
+    private void sortOtherIdentifiersAndUpdateInstance(GKInstance rgpInst) throws Exception {
+        List<String> otherIdentifiers = rgpInst.getAttributeValuesList(ReactomeJavaConstants.otherIdentifier);
+        Collections.sort(otherIdentifiers);
+        adapter.updateInstanceAttribute(rgpInst, ReactomeJavaConstants.otherIdentifier);
+    }
+
+    // Checks that none of the provided mapping structures are null.
+    private boolean allDataStructuresPopulated(Map<String, List<String>> proteinToTranscripts, Map<String, List<String>> transcriptToProbes, Map<String, List<String>> uniprotToProtein) {
+        return proteinToTranscripts != null && transcriptToProbes != null && uniprotToProtein != null;
+    }
+
+    // Get species name from json config file.
+    private Set<String> getSpeciesNames() throws IOException, ParseException {
+        Properties applicationProps = new Properties();
+        String propertiesLocation = System.getProperty("config.location");
+        try(FileInputStream fis = new FileInputStream(propertiesLocation))
+        {
+            applicationProps.load(fis);
+        }
+        String pathToSpeciesConfig = applicationProps.getProperty("pathToSpeciesConfig");
+        JSONParser parser = new JSONParser();
+        Object obj = parser.parse(new FileReader(pathToSpeciesConfig));
+        JSONObject jsonObject = (JSONObject) obj;
+        Set<String> fullSpeciesNames = new HashSet<>();
+        for (Object speciesKey : jsonObject.keySet()) {
+            JSONObject speciesJson = (JSONObject) jsonObject.get(speciesKey);
+            JSONArray speciesNames = (JSONArray) speciesJson.get("name");
+            String speciesName = (String) speciesNames.get(0);
+            fullSpeciesNames.add(speciesName);
+        }
+        return fullSpeciesNames;
```

Comment:
This logic looks similar to the loading of the properties file and JSON parsing in the `EnsemblBiomartRetriever` class.

Could this be extracted to two methods (one for loading the properties file and one for parsing the JSON) in a helper utility class?

Reply:
Done